airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -73,7 +76,41 @@ def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) ->
             params[self.filter_field] = self.start_date
         return params
 
+    @staticmethod
+    def control_request_rate_limit(response: requests.Response, treshold: float = rate_limit_treshold) -> float:
+        """
+        To avoid reaching Shopify API Rate Limits, use the "X-Shopify-Shop-Api-Call-Limit" header value,
+        to determine the current rate limits and load and handle sleep time based on load %.
+        Recomended sleep time between each request is 1 sec, we would handle this dynamicaly.
+        More information: https://shopify.dev/api/usage/rate-limits
+        """
+        # default sleep time between each request - 100 miliseconds
+        sleep_time = 0.1
+        # average load based on treshold
+        avg_load = treshold / 2
+        # Get the rate_limits from response
+        rate_limits = response.headers.get("X-Shopify-Shop-Api-Call-Limit", None)
+
```

Comment:
add an example of values for this header

Reply:
Done.